app/domain/structured_log.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module StructuredLog
+  # See https://github.com/tidelift/design-docs/blob/main/2022-02-backend-logging-conventions.md for conventions
+  #
+  # Timing example:
+  #
+  #  StructuredLog.capture(
+  #    "SOME_TIMING_METRIC",
+  #    {
+  #      class_name: "User",
+  #      method: "a_slow_method",
+  #      some_arbitrary_data: 123,
+  #      ms: 123.45,
+  #    }
+  #  )
+  #
+  # Error occurrence example:
+  #
+  #  StructuredLog.capture(
+  #    "SOME_ERROR_METRIC",
+  #    {
+  #      class_name: "Team",
+  #      method: "some_erroring_method",
+  #      some_arbitrary_data: 123,
+  #      message: "A description of the error",
+  #    }
+  #  )
+  #
+  def self.capture(name, data_hash)
+    rails_logger(name, data_hash)
+  rescue ArgumentError => e
+    Rails.logger.error(e.message)
+    raise e
+  rescue StandardError => e
+    Rails.logger.error("Error capturing structured log for metric=#{name} - #{e.message}")
+  end
+
+  def self.rails_logger(name, data_hash)
```

Comment:
Perhaps something like `def self.log_info` or so?  To me at least, `rails_logger` implies a logging object is the output of this method. Including `info` in there somewhere can make it more descriptive & differentiate it from warning/error logs as well.